### PR TITLE
Move servlet-specifics out of more general http namespaces

### PR DESCRIPTION
**Pull request by ddeaguiar**
This PR addresses #520, following the implementation approach provided by @ohpauleez. Refer to the issue for details.

### CODE REVIEW

File: service/src/io/pedestal/http/request/servlet_support.clj [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  (header [req header-string] (.getHeader req header-string))
+  (ssl-client-cert [req] (.getAttribute req "javax.servlet.request.X509Certificate"))
+  (body [req] (.getInputStream req))
+  (path-info [req] (servlet-path-info req))
+  (async-supported? [req] (.isAsyncSupported req))
+  (async-started? [req] (.isAsyncStarted req)))
+
+(extend-protocol request/ResponseBuffer
+  HttpServletResponse
+  (response-buffer-size [req]))
```
Comment (ohpauleez) [inline service/src/io/pedestal/http/request/servlet_support.clj]: You still need to move `.getBufferSize` in here, correct?

Comment (ddeaguiar) [reply service/src/io/pedestal/http/request/servlet_support.clj]: Yep, missed that :/. I'll rename the param to `resp` as well.

**Comment by ohpauleez** [comment]
Thanks a ton for pulling all these changes together -- they represent all the changes I had in mind.

Going forward, you can just make branches on the Pedestal repo directly (it makes looking at diffs locally a little easier).

One thing that I'd typically do in a changeset like this is take some good profiling and benchmark numbers before and after.  To do that, I look at the following:
 * Profiling info from common/happy path after JIT warm-up (do the changes prevent JIT optimizations? Where are we spending most of our time?  What calls are dominating?  Is there a clear hot spot that needs to be addressed?)
 * Length of the test suite running
 * Throughput/goodput and latency across standard applications (I have a handful of Pedestal applications I test, but you could always use the samples as a stand in if you don't have applications on hand)
 * Throughput/goodput and latency of the fastest Pedestal setups (again, I have some internal stuff I use).
 * Any other benchmarks or inspections that make sense given the changes

**Comment by ddeaguiar** [comment]
@ohpauleez, Good point. In regards to the fastest Pedestal setups, would the [fast-pedestal](https://github.com/pedestal/pedestal/tree/master/samples/fast-pedestal) sample be sufficient or should I consider other setups as well?

On a separate note, It's not clear to me what's causing the integration test failure. It passes on my machine but I noticed that I'm using a different version of Java (1.8.0_112) and lein (2.7.1). Perhaps it's related to that? I'll try upgrading locally to verify.

Thanks for taking the time to review this!

**Comment by ddeaguiar** [comment]
@ohpauleez, I had tried to create the branch in this repo but I don't have access yet. If you or @mtnygard can grant me the requisite privileges I'll do that in the future.

**Comment by ohpauleez** [comment]
@ddeaguiar `fast-pedestal` sample would be sufficient, in both the `fast-jetty` and `faster-jetty` setup, using both `wrk` and `wrk2` for getting numbers.  If you have a real Pedestal project that uses a highly optimized chain provider (like a service router or something), that's also usually a better source for comparative testing because it actually performs some sort of business logic.

I'll let @mtnygard sort out the perms for you.

Ping me here if you need an extra set of eyes to sort out the integration issue.

**Comment by ddeaguiar** [comment]
@ohpauleez, the ci builds are failing because the server-integration-test [depends on *file*](https://github.com/pedestal/pedestal/blob/master/service-template/test/pedestal/new_server_integration_test.clj#L23) which is nil with lein 2.8.1. Ironically these tests are not intended to be run in the ci build (hence the usage of `lein test :travis`). I'll spend some time looking into this but I'm open to suggestions.

**Comment by ddeaguiar** [comment]
Ping @mtnygard, @ohpauleez. I updated with `master` and CI is now green. I ran some benchmarks with `wrk` and `wrk2` against the `faster-jetty` sample and did not observe a performance impact. I think this PR is ready to merge.

I'm going to look into automating the performance test verification.

**Comment by ohpauleez** [comment]
I'm merging this.  It'll clean up a handful of branches I've got in-flight.

**Pull request merged**